doc/development.rst

```diff
@@ -172,3 +172,186 @@ If there are open questions, final decisions are made by the Temporary
 Benevolent Dictator, currently Gaël Varoquaux.
 
 .. include:: ../CONTRIBUTING.rst
+
+How to make a release?
+======================
+
+This section describes how to make a new release of Nilearn.
+We assume that we are in a clean state where all the Pull Requests (PR) that we wish to include in the new release have been merged.
+For example, make sure all deprecations that are supposed to be removed with this new version have been adressed. Furthermore, if this new release comes with dependency version bumps (Python, Numpy...), make sure to implement and test these changes beforehand. Ideally, these would have been done before such as to update the code base if necessary. Finally, make sure the documentation can be built correctly.
+
+Prepare the release
+-------------------
+
+First we need to prepare the release by updating the file `nilearn/doc/whats_new.rst` to make sure all the new features, enhancements, and bug fixes are included in their respective sections.
+We also need to write a "Highlights" section promoting the most important additions that come with this new release, and add the version tag just above the corresponding title:
+
+.. code-block:: RST
+
+    .. _vx.y.z:
+
+    x.y.z
+    =====
+    **Released MONTH YEAR**
+
+    HIGHLIGHTS
+    ----------
+    
+    - Nilearn now includes functionality A
+    - ...
+
+
+Next, we need to bump the version number of Nilearn by updating the file `nilearn/version.py` with the new version number, that is edit the line:
+
+.. code-block:: python
+
+    __version__ = x.y.z.dev
+
+
+to be:
+
+.. code-block:: python
+   
+    __version__ = x.y.z
+
+
+We also need to update the website news section by editing the file `nilearn/doc/themes/nilearn/layout.html`. The news section typically contains links to the last 3 releases that should look like:
+
+.. code-block:: html
+    
+    <h4> News </h4>
+        <ul>
+            <li><p><strong>November 2020</strong>: 
+                <a href="whats_new.html#v0-7-0">Nilearn 0.7.0 released</a>
+            </p></li>
+            <li><p><strong>February 2020</strong>: 
+                <a href="whats_new.html#v0-6-2">Nilearn 0.6.2 released</a>
+            </p></li>
+            <li><p><strong>January 2020</strong>: 
+                <a href="whats_new.html#v0-6-1">Nilearn 0.6.1 released</a>
+            </p></li>
+        </ul>
+
+
+Here, we should remove the last entry and add the new release on top of the list.
+
+In addition, we can have a look at `MANIFEST.in` to check that all additional files that we want to be included or excluded from the release are indicated. Normally we shouldn't have to touch this file.
+
+Add these changes and the version tag, and commit them:
+
+.. code:: bash
+    
+    git add doc/whats_new.rst nilearn.version.py
+    git commit -m "REL x.y.z"
+    git tag x.y.z
+    git push origin release-x.y.z
+    git push --tags
+
+
+Build the distributions and upload them to Pypi
```

Comment:
as this matches closely the instructions from https://packaging.python.org/guides/distributing-packages-using-setuptools/#id70, maybe add a link?